compiler-core/src/language_server/engine.rs

```diff
@@ -520,14 +520,44 @@ fn hover_for_function_head(
 ) -> Hover {
     let empty_str = EcoString::from("");
     let documentation = fun.documentation.as_ref().unwrap_or(&empty_str);
-    let function_type = Type::Fn {
-        args: fun.arguments.iter().map(|arg| arg.type_.clone()).collect(),
-        retrn: fun.return_type.clone(),
-    };
-    let formatted_type = Printer::new().pretty_print(&function_type, 0);
+
+    let mut function_string = format!("fn {}(", fun.name);
```

Comment:
FWIW https://github.com/gleam-lang/gleam/pull/2705/files it looks like there is an existing draft pr that utilizes the function used by the doc generation to do roughly the same thing https://github.com/gleam-lang/gleam/blob/main/compiler-core/src/format.rs#L1403 not sure if theres a preference to reuse vs write new here but worth considering

Reply:
@Acepie Oh nice, I think it's targeting same set of changes. I wasn't aware of this `format`. Looks like that way is better. I will discuss on that PR 👍🏽 Thanks.